src/app/app.authorization.js

```diff
@@ -0,0 +1,13 @@
+angular.module('app').run([
+  '$rootScope', '$state', 'authenticationService', '$http', 'localStorageService',
+  function ($rootScope, $state, authenticationService) {
+
+  $rootScope.$on('$stateChangeStart',
+    function (event, toState, toParams, fromState, fromParams) {
+      if (!authenticationService.isLoggedIn() && toState.data.requireLogin) {
```

Comment:
good question.. now I see why there's this validation in the authorization that you've written...

else if (authenticationService.isLoggedIn() && !toState.data.requireLogin)

however, in my training app, I have some states that doesnt require login and I dont want to "prevent default" in them if someone that is logged in enters.. one example of that is "home" state.. If someone goes to "home" I want them to see it, even if they are / are not logged in... How should I handle this? 
I thought in somethink like

else if (authenticationService.isLoggedIn() && !toState.url==='/login')
event.preventDefault();
alert('NOPE, you are already logged')
@sbalay , is my approach correct? :S